apps/emqx_bridge/include/emqx_bridge_v2_register.hrl

```diff
@@ -0,0 +1,153 @@
+%% This function is called to register a bridge V2. It should be called before
+%% the system boots in a function triggered by an -on_load() directive
+%% since it should be called before the system boots because the config
+%% system depends on that.
+%%
+%% It is placed in an hrl file instead of in emqx_bridge_v2.erl because emqx_bridge_v2
+%% might not be loaded when the bridge module is loaded.
+-spec emqx_bridge_v2_register_bridge_type(#{
+    %% Should be provided by all bridges. Even if the bridge_v2_type_name is
+    %% the same as the bridge_v1_type_named.
+    'bridge_v1_type_name' := atom(),
+    'bridge_v2_type_name' := atom(),
+    'connector_type' := atom(),
+    'schema_module' := atom(),
+    'schema_struct_field' := atom() | binary()
+}) -> ok.
+emqx_bridge_v2_register_bridge_type(BridgeTypeInfo) ->
+    try
+        %% We must prevent overwriting so we take a lock when writing to persistent_term
+        global:set_lock(
+            {
+                internal_emqx_bridge_v2_persistent_term_info_key(),
+                internal_emqx_bridge_v2_persistent_term_info_key()
+            },
```

Comment:
Just thinking out loud: I wonder if we could avoid locking if each module has its own exclusive key (e.g.: `{common_info_key(), bridge_type()}`), and then coalesced to a single value when starting the `emqx_bridge` up.

Reply:
We could do that but I think that could be a future improvement in that case. Unless the single node locking in `global` is extremely expensive (which I don't see any reason for it to be) the performance benefits if any should be very minimal. Also, the the locked critical section is so short so I think the the wait on lock overhead will be minimal even when we have refactored all 20 bridges (I'm not even sure module loading is happening in parallel).